File: software/examples/c/example_callback.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_gps.h"

#define HOST "localhost"
#define PORT 4223
#define UID "ABC" // Change to your UID

// Callback function for current callback (parameter has unit mA)
void cb_coordinates(uint32_t latitude, char ns, uint32_t longitude, char ew,
                    uint16_t pdop, uint16_t hdop, uint16_t vdop, uint16_t epe) {
	printf("Latitude: %f ° %c\n", latitude/1000000.0, ns);
	printf("Longiutde: %f ° %c\n", longitude/1000000.0, ew);
}

int main() {
	// Create ip connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	GPS gps;
	gps_create(&gps, UID); 

	// Add device to ip connection
	if(ipcon_add_device(&ipcon, &gps) < 0) {
		fprintf(stderr, "Could not connect to Brick\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Set Period for coordinates callback to 1s (1000ms)
	// Note: The callback is only called every second if the 
	//       coordinates have changed since the last call!
	gps_set_coordinates_callback_period(&gps, 1000);

	// Register coordinates callback to function cb_coordinates
	gps_register_callback(&gps, GPS_CALLBACK_COORDINATES, cb_coordinates);

	printf("Press ctrl+c to close\n");
	ipcon_join_thread(&ipcon); // Join mainloop of ip connection
}

File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletGPS.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletGPS;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your GPS Bricklet

// Callback function for coordinates callback
function cb_coordinates($latitude, $ns, $longitude, $ew, $pdop, $hdop, $vdop, $epe)
{
    echo "Latitude: " . $latitude/1000000.0 . " °\n";
    echo "N/S: $ns\n";
    echo "Longitude: " . $longitude/1000000.0 . " °\n";
    echo "E/W: $ew\n";
    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$gps = new BrickletGPS(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register coordinates callback to function cb_coordinates
$gps->registerCallback(BrickletGPS::CALLBACK_COORDINATES, 'cb_coordinates');

// Set period for coordinates callback to 1s (1000ms)
// Note: The coordinates callback is only called every second
//       if the coordinates has changed since the last call!
$gps->setCoordinatesCallbackPeriod(1000);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
